falcon/request_helpers.py

```diff
@@ -38,6 +42,29 @@ def fget(self):
     return property(fget)
 
 
+def parse_etags(etag_str):
+    """
+    Parse a string of ETags given in an If-Match or If-None-Match header as
+    defined by RFC 7232.
+
+    Args:
+        etag_str (str): A string of ASCII characters placed between double quotes
+            and may be prefixed by ``W/`` to indicate that the weak comparison
+            algorithm should be used.
+
+    Returns:
+        A list of unquoted ETags or ``['*']`` if all ETags should be matched.
+    """
+    if not etag_str:
+        return []
+
+    if etag_str.strip() == '*':
+        return ['*']
+
+    matches = re.findall(_ETAG_PATTERN, etag_str.strip())
```

Comment:
PERF: It may improve performance to short-circuit and avoid using a regex for the common case of a single etag. Something like:

1. If the string does not contain a comma, go to step 2.
2. If the string starts with `W/"` assume it also ends in `"` and do a quick string slice to produce the result. Otherwise, go to step 3.
3. If the string starts with `"`, assume it also ends in `"` and return the string slice. Otherwise go to step 4.
4. The string is unquoted. But since we are nice, we will just return the string as-is rather than returning an empty list. 

To be strictly correct, we could also verify that strings that start with quotes do indeed also end with them, but IMHO if the etag is malformed in that way, there is something seriously broken downstream and it shouldn't be Falcon's job to work around it.